### PR TITLE
Added URL update on scroll

### PR DESCRIPTION
**Pull request by morkro**
This updates the hash in the URL when scrolling the site with the current visible section's ID.

### CODE REVIEW

File: assets/js/src/helpers.js [hunk 1]
```diff
@@ -20,4 +20,4 @@ export default {
 
     return offset;
   }
-};
+};
```
Comment (HugoGiraudel) [inline assets/js/src/helpers.js]: Missing EOF.

File: assets/js/src/main.js [hunk 2]
```diff
@@ -3,4 +3,5 @@
 import '../vendor/bling';
 import '../vendor/accessible-modal-dialog';
 import './sidebar';
-import './app';
+import './url';
+import './app';
```
Comment (HugoGiraudel) [inline assets/js/src/main.js]: Missing EOF.

File: assets/js/src/url.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/* globals S */
+
+import h from './helpers';
+
+export default (function () {
+  'use strict';
+
+  // DOM queries
+  var headings = $('.chapter:not(.toc) > h1[id]');
+  // Internal variables
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: Missing empty line.

File: assets/js/src/url.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/* globals S */
+
+import h from './helpers';
+
+export default (function () {
+  'use strict';
+
+  // DOM queries
+  var headings = $('.chapter:not(.toc) > h1[id]');
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: I think we can leave `.toc` here: `$('.chaper > h1[id]')`.

File: assets/js/src/url.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  // Internal variables
+  var offset = 50;
+  var headingsOffset = headings.map(function(el) {
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: Missing space before parenthese.

File: assets/js/src/url.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    return [h.getOffset(el), el.id];
+  });
+
+  var updateURL = function () {
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: We use function declarations rather than function expressions.

Comment (HugoGiraudel) [inline assets/js/src/url.js]: I wonder whether we should make this part of sidebar.js since they are highly similar?

File: assets/js/src/url.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+      if (headingPassed) {
+        setHash(headingsOffset[i][1]);
+      } else if (lastHeading) {
+        setHash(headingsOffset[headingsOffset.length - 1][1]);
+      }
+    }
+  };
+
+  var setHash = function (id) {
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: Function declaration again.

File: assets/js/src/url.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    if (!id && !arguments.length) return;
+    window.history.pushState({}, '', '#' + id);
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: So you ended up using pushState?

File: assets/js/src/url.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      if (headingPassed) {
+        setHash(headingsOffset[i][1]);
+      } else if (lastHeading) {
+        setHash(headingsOffset[headingsOffset.length - 1][1]);
+      }
+    }
+  };
+
+  var setHash = function (id) {
+    if (!id && !arguments.length) return;
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: Why do you check `arguments`?

File: assets/js/src/url.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    window.history.pushState({}, '', '#' + id);
+  }
+
+  document.addEventListener('scroll', updateURL, false);
+
+  // Initial exec
+  updateURL();
+}());
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: Missing EOF.

File: assets/js/src/url.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
+/* globals S */
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: `$`

File: assets/js/src/url.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+      } else if (lastHeading) {
+        setHash(headingsOffset[headingsOffset.length - 1][1]);
+      }
+    }
+  };
+
+  var setHash = function (id) {
+    if (!id && !arguments.length) return;
+    window.history.pushState({}, '', '#' + id);
+  }
```
Comment (HugoGiraudel) [inline assets/js/src/url.js]: Missing `;`.

**Comment by HugoGiraudel** [comment]
WTF is happening here @morkro? :D

**Comment by morkro** [comment]
Haha, I completely forgot about this one. You added all your feedback and then... I just stopped? Let me look into that branch again.

**Comment by HugoGiraudel** [comment]
Classic @morkro. <3

**Comment by HugoGiraudel** [comment]
So, shall I close or? :D

**Comment by morkro** [comment]
I think we have changed so much over the past year, I would've to start over anyway. Maybe keep the branch so we don't lose all the effort yet?

**Pull request closed without merging**